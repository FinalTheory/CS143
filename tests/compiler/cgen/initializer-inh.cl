class A inherits IO {
  o : Object <- f();
  x : Int <- 12;
  f() : Object {{
    out_string("A.f: x = ");out_int(x);out_string("\n");
  }};
};

class B inherits A {
  y : Int <- {
    out_string("initializing B.y. here x = ");out_int(x);out_string("\n");
    13;};
  f() : Object {{
    out_string("B.f: y = ");out_int(y);out_string("\n");
  }};
};

class Main inherits IO {
  main() : Object {{
    out_string("new A\n");
    new A;
    out_string("new B\n");
    new B;
  }};
};
