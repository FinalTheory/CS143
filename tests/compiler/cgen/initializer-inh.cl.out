SPIM Version 6.5 of January 4, 2003
Copyright 1990-2003 by James R. Larus (larus@cs.wisc.edu).
All Rights Reserved.
See the file README for a full copyright notice.
Loaded: /usr/class/cs143/cool/lib/trap.handler
new A
A.f: x = 0
new B
B.f: y = 0
initializing B.y. here x = 12
COOL program successfully executed
Stats -- #instructions : 590
         #reads : 127  #writes 91  #branches 109  #other 263
